### PR TITLE
fix: pin with `exact=False`

### PR DESCRIPTION
**Pull request by ltalirz**
Pinning with `exact=True` requires the build strings of the packages to be identical. Looking at the package builds at

  https://anaconda.org/conda-forge/aiida-core/files
  https://anaconda.org/conda-forge/aiida-core.atomic_tools/files
  https://anaconda.org/conda-forge/aiida-core.services/files

we find that the build string of `aiida.core` and the other outputs differs.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Review by chrisjsewell — CHANGES_REQUESTED** [review]
I disagree with this change

**Comment by ltalirz** [comment]
Perhaps I'm wrong but as far as I understand, the problem is that, for release 2.2.2,  the [build string of the `aiida-core` package](https://anaconda.org/conda-forge/aiida-core/files) is pyh1a96a4e_0 while the build string of `aiida-core.services` (and the other outputs) is `hd8ed1ab_0`

and that the exact pin therefore prevents them from being installed into the same environment

**Comment by chrisjsewell** [comment]
Yeh, you haven't explained very well why this can't be used. 

The dependencies should all come from the same release; this is the version plus the build number, if the build number is different, then there is likely a good reason for that, e.g. dependency pinnings have been changed.

After a quick google, I found this: https://github.com/conda/conda-build/issues/4142.
Following the links, this seems to have been fixed in https://github.com/mamba-org/boa/pull/249 (which conda-forge uses),
but perhaps its not rolled out yet or something 🤷 (can you have a look)

TLDR, we should be using exact `equals=True`, but perhaps can't right now do to a bug in conda.

**Comment by chrisjsewell** [comment]
If we do make the change, then we should make an issue to put it back, when conda-forge fixes the issue

**Comment by ltalirz** [comment]
> If we do make the change, then we should make an issue to put it back, when conda-forge fixes the issue

That's fine with me.

I do think it's important to make sure that one can `conda install -c conda-forge aiida` (or `aiida-core aiida-core.services`), so this fix is rather urgent - agreed?

**Comment by chrisjsewell** [comment]
Yep, but make the issue first 😉

**Comment by ltalirz** [comment]
> The dependencies should all come from the same release; this is the version plus the build number, if the build number is different, then there is likely a good reason for that, e.g. dependency pinnings have been changed.

I don't know why the build strings are different / why the `aiida-core` build string starts with `py` -  perhaps because `aiida-core` is the only package with `python` explicitly in the `host:` requirements?
If you look at the links I shared, this is not something that changed recently, it has been like this all along

**Comment by chrisjsewell** [comment]
yeh the build string is essentially encapsulating build number + python version, but unfortunately we only want the same build_number

File: recipe/meta.yaml [hunk 1]
```diff
@@ -112,7 +112,7 @@ outputs:
       noarch: python
     requirements:
       run:
-        - {{ pin_subpackage(aiidacore, exact=True) }}
```
Comment (chrisjsewell) [inline recipe/meta.yaml:115]: given the comments, I think removing`exact=True` is not necessary for python extras, since these also have a python version in the build string, so will match

**Comment by ltalirz** [comment]
> Yep, but make the issue first 😉

done!

File: recipe/meta.yaml (hunk 1 above)
Comment (ltalirz) [reply recipe/meta.yaml:115]: No, as you can see here, even those have the build string that does not start with `py` https://anaconda.org/conda-forge/aiida-core.atomic_tools/files

Perhaps because there is no explicit dependency on python

Comment (chrisjsewell) [reply recipe/meta.yaml:115]: i.e from the same build `aiida-core` and `aiida-core.atomic_tools` etc will always have the same build string, but because `aiida-core.services` does not include python it will not

Comment (chrisjsewell) [reply recipe/meta.yaml:115]: I would maybe the put the python dependency in these also, which should then give the same build string

Comment (ltalirz) [reply recipe/meta.yaml:115]: Ok, let's try it. I will still remove `exact=True`.

If it turns out that the build strings are identical, we can then reintroduce the pin as before

Comment (chrisjsewell) [reply recipe/meta.yaml:115]: Ok, never say I don't compromise lol

**Comment by ltalirz** [comment]
ok, looks like that did the trick

```
BUILD START: ['aiida-core-2.2.2-pyh1a96a4e_1.conda', 'aiida-core.services-2.2.2-hd8ed1ab_1.conda', 'aiida-2.2.2-hd8ed1ab_1.conda', 'aiida-core.atomic_tools-2.2.2-pyh1a96a4e_1.conda', 'aiida-core.notebook-2.2.2-pyh1a96a4e_1.conda', 'aiida-core.rest-2.2.2-pyh1a96a4e_1.conda']
```

I'll add python to the `aiida` metapackage as well and reintroduce the pins, except for the one on `aiida-core.services`.

**Review by chrisjsewell — APPROVED** [review]
cheers

**Comment by ltalirz** [comment]
For the record, these are the expected build strings:
```
2023-02-13T13:37:37.5304476Z BUILD START: ['aiida-core-2.2.2-pyh1a96a4e_1.conda', 'aiida-core.services-2.2.2-hd8ed1ab_1.conda', 'aiida-2.2.2-pyh1a96a4e_1.conda', 'aiida-core.atomic_tools-2.2.2-pyh1a96a4e_1.conda', 'aiida-core.notebook-2.2.2-pyh1a96a4e_1.conda', 'aiida-core.rest-2.2.2-pyh1a96a4e_1.conda']
```
